packages/kuma-gui/src/app/resources/data/Resource.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { describe, expect, test } from 'vitest'
+
+import { Resource } from './Resource'
+
+describe('Resource.search', () => {
+  describe('commons', () => {
+    test.each([
+    // Name
+      [
+        'foo',
+        {
+          name: 'foo',
+        },
+      ],
+      [
+        'name:foo',
+        {
+          name: 'foo',
+        },
+      ],
+
+      // Shortcuts
+      [
+        'namespace:foo',
+        {
+          ['filter[labels.k8s.kuma.io/namespace]']: 'foo',
+        },
+      ],
+      [
+        'zone:foo',
+        {
+          ['filter[labels.kuma.io/zone]']: 'foo',
+        },
+      ],
+      [
+        'service:foo',
+        {
+          ['filter[labels.kuma.io/service]']: 'foo',
+        },
+      ],
+      [
+        'namespace:foo service:bar zone:baz',
+        {
+          ['filter[labels.k8s.kuma.io/namespace]']: 'foo',
+          ['filter[labels.kuma.io/service]']: 'bar',
+          ['filter[labels.kuma.io/zone]']: 'baz',
+        },
+      ],
+
+      // Labels
+      [
+        'foo:bar',
+        {
+          ['filter[labels.foo]']: 'bar',
+        },
+      ],
+      [
+        'label:foo:bar',
+        {
+          ['filter[labels.foo]']: 'bar',
+        },
+      ],
+      [
+        'label: foo:bar',
+        {
+          ['filter[labels.foo]']: 'bar',
+        },
+      ],
+      [
+        'label:foo:bar label:baz=qux label: quux:quuux label: kuma.io/service=foo',
+        {
+          ['filter[labels.foo]']: 'bar',
+          ['filter[labels.baz]']: 'qux',
+          ['filter[labels.quux]']: 'quuux',
+          ['filter[labels.kuma.io/service]']: 'foo',
+        },
+      ],
+
+      // Invalids
+      [
+        ':',
+        {},
+      ],
+      [
+        'foo:',
+        {},
+      ],
+      [
+        ':foo',
+        {},
+      ],
+    ])('%s works', (search, expected) => {
+      const actual = Resource.search(search)
+      expect(actual).toStrictEqual(expected)
+    })
+  })
+
+  describe('defaultKey', () => {
+    for(const defaultKey in ['foo', 'bar']) {
```

Comment:
I think its a bit weird that we are using a `test.each` and a `for` loop? Should we just make two test cases in the `test.each` by altering the args for the test a little